target/linux/pistachio/base-files/etc/init.d/bt-csr

```diff
@@ -139,22 +139,30 @@ EOF
 		}
 
 	fi
-
-# Configure HCI and get some version info
-	BT_ADDR_MASK=10
-	hciconfig hci0 reset &&\
-	hciconfig hci0 up &&\
-	hciconfig hci0 piscan &&\
-	hciconfig hci0 version &&\
-	BT_ADDR=$(echo $MAC | cut -c $BT_ADDR_MASK-) &&\
-	BT_NAME="Creator-Ci40["$BT_ADDR"]" &&\
-	hciconfig hci0 name $BT_NAME &&\
-	hcitool dev || (echo  "Failed to configure" && exit 1)
-
 }
 
+
 start() {
+	. /lib/functions/bluetooth.sh
+
 	echo  "Starting Bluetooth..."
+	local timeout=10
+	local time=`awk -F "." '{print $1}' /proc/uptime | awk '{print $1}'`
+	local elap_time=0
+
+	lsmod | grep -q hci_uart
```

Comment:
-q option will take care of the output not appearing in stdout, redirecting was causing the false condition to be true as the redirection succeeds the $? will be 0 even if the module is missing